### PR TITLE
CMOS-158: Update example docs

### PR DESCRIPTION
**Pull request by patrick-stephens**
Add optional container build to the `run.sh` script.
The `Makefile` target does run the `container` target first though anyway.

### CODE REVIEW

File: examples/containers/run.sh [hunk 1]
```diff
@@ -22,8 +22,12 @@ DOCKER_USER=${DOCKER_USER:-couchbase}
 DOCKER_TAG=${DOCKER_TAG:-v1}
 CMOS_IMAGE=${CMOS_IMAGE:-$DOCKER_USER/observability-stack:$DOCKER_TAG}
 
-# Ensure we built the container locally first
-make -C "${SCRIPT_DIR}/../.." container
+# Ensure we build the container locally first otherwise make
+# sure one is tagged as above CMOS_IMAGE for use in the .env file.
+if [[ "${SKIP_CONTAINER_BUILD:-yes}" != "yes" ]]; then
+    echo "Building CMOS container"
+    make -C "${SCRIPT_DIR}/../.." container
```
Comment (redm2) [inline examples/containers/run.sh:29]: So the non-oss is effectively the default. What's a possible entrypoint where the container isn't locally built - if run.sh is run directly?

File: examples/containers/run.sh [hunk 2]
```diff
@@ -20,14 +20,9 @@ COUCHBASE_SERVER_IMAGE=${COUCHBASE_SERVER_IMAGE:-couchbase/server:7.0.2}
 
 DOCKER_USER=${DOCKER_USER:-couchbase}
 DOCKER_TAG=${DOCKER_TAG:-v1}
-CMOS_IMAGE=${CMOS_IMAGE:-$DOCKER_USER/observability-stack:$DOCKER_TAG}
-
 # Ensure we build the container locally first otherwise make
```
Comment (redm2) [inline examples/containers/run.sh:23]: Left in?

Comment (patrick-stephens) [reply examples/containers/run.sh:23]: It's required to set the variable to compose if you do want to override it. Just making it explicit what is important and also the defaults.

Comment (redm2) [reply examples/containers/run.sh:23]: Sorry I mean the comment "# Ensure we build the container locally first..."

Comment (patrick-stephens) [reply examples/containers/run.sh:23]: Yeah I wanted to document it here as well, someone might just look at the run.sh script.

Comment (redm2) [reply examples/containers/run.sh:23]: Ah i see, maybe rewrite as "Ensure you have built the container... "

**Review by redm2 — APPROVED** [review]
Looks good.

**Pull request merged**